script.js

```diff
@@ -26,18 +26,98 @@ function createProductItemElement({ sku, name, image }) {
+  selectDiv.innerHTML = '';
+  createH2.innerHTML = await sumReturn;
+  selectDiv.appendChild(createH2);
+};
+
+const fetchCarItems = async (itemID) => {
+  const product = await fetch(`https://api.mercadolibre.com/items/${itemID}`);
+  const response = await product.json();
+  await appendValue(sumItemsPrices());
+  return response;
+};
 
 function cartItemClickListener(event) {
-  // coloque seu código aqui
-}
+  const getTargetId = event.target.id;
+  const response = fetchCarItems(getTargetId);
+  const getOL = document.getElementsByClassName('cart__items')[0];
+  getOL.removeChild(event.target);
+  cartItemsList.forEach((item, index) => {
+    if (item.id === getTargetId) cartItemsList.splice(index, 1);
+    localStorage.setItem('carShop', JSON.stringify(cartItemsList));
+  });
+  return response;
+ }
+
+ // Fonte para eliminar um item de uma array: https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/splice
 
 function createCartItemElement({ sku, name, salePrice }) {
   const li = document.createElement('li');
   li.className = 'cart__item';
+  li.id = `${sku}`;
   li.innerText = `SKU: ${sku} | NAME: ${name} | PRICE: $${salePrice}`;
   li.addEventListener('click', cartItemClickListener);
   return li;
 }
+
+const createCartItems = async () => {
+  const carOL = document.getElementsByClassName('cart__items')[0];
+  carOL.innerHTML = '';
+  await appendValue(sumItemsPrices());
+  cartItemsList.forEach((itm) => {
+    const creatLi = createCartItemElement({ sku: itm.id, name: itm.title, salePrice: itm.price });
+    carOL.appendChild(creatLi);
+  });
+};
+  // Tive ajuda do instrutor Eduardo para finalizar o requisito 1.
+const renderItems = (getResults) => {
+  let pc;
+  const selectSection = document.getElementsByClassName('items')[0];
+  getResults.forEach((el) => {
+   pc = createProductItemElement({ sku: el.id, name: el.title, image: el.thumbnail });
+    selectSection.appendChild(pc);
+    pc.addEventListener('click', () => {
+      const carProduct = cartItemsList.find((item) => item.id === el.id);
+      if (!carProduct) cartItemsList.push(el);
+      localStorage.setItem('carShop', JSON.stringify(cartItemsList));
+      createCartItems();      
+    });
+   });
+  };
+
+  const emptyCart = async () => {
+    const selectbtn = document.getElementsByClassName('empty-cart')[0];
+     selectbtn.addEventListener('click', () => {
+      cartItemsList = [];
+      localStorage.removeItem('carShop');
+      createCartItems();
+    });
+  };
+  
+window.onload = async function onload() { 
+  console.log('Ok!!! Ready');
+  const getResult = await listOfProducts();
+  const getLoadEl = document.querySelector('.loading');
+  getLoadEl.parentNode.removeChild(getLoadEl);
+  renderItems(getResult);
+  createCartItemElement(getResult);
+  const itemsStorageString = localStorage.getItem('carShop');
+  cartItemsList = itemsStorageString ? JSON.parse(itemsStorageString) : [];
```

Comment:
```suggestion
  cartItemsList = itemsStorageString ? JSON.parse(itemsStorageString) : [];
```
Boa! Muitas pessoas esqueceram desse tipo de verificação o que gerava problemas mais pra frente no código.